ckanext/canada/logic.py

```diff
@@ -185,3 +195,105 @@ def datastore_create_temp_user_table(context):
         '''.format(
             username=literal_string(username),
             sysadmin='TRUE' if is_sysadmin(username) else 'FALSE'))
+
+
+@chained_action
+def canada_resource_view_show(up_func, context, data_dict):
+    """
+    Return the metadata of a resource_view.
+
+    :param id: the id of the resource_view
+    :type id: string
+
+    :rtype: dictionary
+
+    Canada Fork: extends the resource_view_show action method
+    to prevent showing datatable_views for invalid and inactive Resources
+
+    We need this method to 404 the resource_view page (e.g. viewing in full screen)
+
+    If a user is logged in, we want them to be able to see and edit the datatables_view
+    views still. We will just add a key to the view dict to be used within templates for visuals.
+    """
+    view_dict = up_func(context, data_dict)
+    if view_dict.get('view_type') == 'datatables_view':
+        # at this point, the core function has been called, calling resource_show
+        # and adding the Resource Object into the context
+        res_extras = getattr(context.get('resource'), 'extras', {})
```

Comment:
I believe this changes upstream, we've stopped caching resource etc. in the context dict because it's unsafe when contexts are passed around and just as fast to rely on the sqlalchemy cache

Reply:
Yeahhhh all these object in the context passed between methods have always confused me anyways. Okay well, same thing, I guess at that point we can assume the Resource exists so I will just use `model.Resource.get`